keps/sig-network/4410-k8s-network-interface/README.md

```diff
@@ -0,0 +1,102 @@
+# KEP-4410: Kubernetes Networking reImagined
+
+> **NOTE**: for the initial PR we've removed a lot of the templated text and
+> aimed to keep this first iteration small and easier to consume. We are only
+> focusing on the "What" and "Why" (e.g. motivation, goals, user stories) for
+> this iteration so that we can build consensus on those first before we add
+> any of the "How".
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (Optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+<!-- /toc -->
+
+## Summary
+
+This proposal is to design and implement the KNI [Kubernetes Networking Interface] or better known as Kubernetes Networking reImagined. KNI will create a Network resource and provide an API that will provide network status, availability, how to attach a pod to a network, detach the pod from the network and update a pods network.  
```

Comment:
I don't think it's "better" known as anything, given that it mostly does not exist yet.

The Summary and Motivation should not assume that the reader is already familiar with the idea.

Reply:
It is funny you mention this because reImagined is how it got recirculated back to me and I just ran with it. However, 100% the summary/motivation should be written in a way that a nontechnical reader should understand. However, that is a high bar to hit as the current is very difficult and through the years, I have found that small numbers of people can actually articulate the current accurately.

Reply:
```suggestion
This proposal is to design and implement a KNI [Kubernetes Networking Interface]. KNI will create a Network resource and provide an API that will provide network status, availability, how to attach a pod to a network, detach the pod from the network and update a pods network.  
```

Reply:
This is my first time trying to get familiar with the KNI project, so started reading this, Apologies if this sounds like a total newbie question, perhaps I must read other material first before coming to this KEP and if so please point me the right way:
```
provide an API that will provide network status, availability, how to attach a pod to a network, detach the pod from the network and update a pods network
```
`network status and availability` means the cluster networking status? node networking status? pod networking status?

the second part refers to pod aspects:
`how to attach a pod to a network, detach the pod from the network and update a pods network` => this is like providing what the CNI spec does but through an API right?